src/main/java/org/terasology/alterationEffects/boost/HealthBoostAlterationEffect.java

```diff
@@ -1,160 +1,75 @@
-            hbot.boostAmount = TeraMath.floorToInt(magnitude);
-            entity.addComponent(hbot);
-        } else {
-            // Remove the current health boost in place.
-            removeBoost(entity, hbot);
-            hbot.boostAmount = TeraMath.floorToInt(magnitude);
-        }
-
-        // Send out this event to collect all the duration and magnitude modifiers and multipliers that can affect this
-        // health boost effect.
-        OnEffectModifyEvent effectModifyEvent = entity.send(new OnEffectModifyEvent(instigator, entity, 0, 0, this, ""));
-        long modifiedDuration = 0;
-        boolean modifiersFound = false;
-
-        // If the effect modify event is consumed, don't apply this health boost effect.
-        if (!effectModifyEvent.isConsumed()) {
-            /*
-            Get the magnitude result value and the shortest duration, and assign them to the modifiedMagnitude and
-            modifiedDuration respectively.
-
-            The shortest duration is used as the effect modifier associated with that will expire in the shortest
-            amount of time, meaning that this effect's total magnitude and next remaining duration will have to be
-            recalculated.
-            */
-            float modifiedMagnitude = effectModifyEvent.getMagnitudeResultValue();
-            modifiedDuration = effectModifyEvent.getShortestDuration();
-
-            // If there's at least one duration and magnitude modifier, set the effect's boost amount and the
-            // modifiersFound flag.
-            if (!effectModifyEvent.getDurationModifiers().isEmpty() && !effectModifyEvent.getMagnitudeModifiers().isEmpty()) {
-                hbot.boostAmount = (int) modifiedMagnitude;
-                modifiersFound = true;
-            }
-        }
+    protected HealthBoostComponent upsertComponent(Optional<HealthBoostComponent> maybeComponent,
+                                                   EffectContext context) {
+        maybeComponent.ifPresent(c -> removeBoost(context.entity, c));
+        HealthBoostComponent component = maybeComponent.orElse(new HealthBoostComponent());
+        component.boostAmount = TeraMath.floorToInt(context.magnitude);
+        component.lastUseTime = time.getGameTimeInMs();
+        return component;
+    }
 
-        hbot.lastUseTime = time.getGameTimeInMs();
+    @Override
+    protected HealthBoostComponent updateComponent(OnEffectModifyEvent event, HealthBoostComponent component,
+                                                   EffectContext context) {
+        component.boostAmount = TeraMath.floorToInt(event.getMagnitudeResultValue());
+        component.lastUseTime = time.getGameTimeInMs();
 
         // Get the health component of this entity, and increase its max health using the health boost multiplier.
-        HealthComponent h = entity.getComponent(HealthComponent.class);
-        h.maxHealth = Math.round(h.maxHealth * (1 + 0.01f * hbot.boostAmount));
```

Comment:
There is no hook to act in this position in the component-based effect. I don't see a nice way to handle this, except for doing these three steps in both `update` and `upsert`

1. revert effect
2. update component
3. apply effect